src/python/pants/backend/go/util_rules/build_pkg_target.py

```diff
@@ -179,8 +198,31 @@ async def setup_build_go_package_target_request(
         objc_files = _first_party_pkg_analysis.m_files
         fortran_files = _first_party_pkg_analysis.f_files
 
+        # If the xtest package was requested, then replace analysis with the xtest values.
+        if request.for_xtests:
+            import_path = f"{import_path}_test"
+            pkg_name = f"{pkg_name}_test"
+            imports = set(_first_party_pkg_analysis.xtest_imports)
+            go_file_names = _first_party_pkg_analysis.xtest_go_files
+            s_files = ()
+            cgo_files = ()
+            cgo_flags = CGoCompilerFlags(
+                cflags=(),
+                cppflags=(),
+                cxxflags=(),
+                fflags=(),
+                ldflags=(),
+                pkg_config=(),
+            )
+            c_files = ()
+            cxx_files = ()
+            objc_files = ()
+            fortran_files = ()
```

Comment:
Why this? These files will never be in the xtest package?

Reply:
Because all of these variables would have been set for the base package in the lines preceding this `if` clause.

Reply:
Right, but how should we reason about whether e.g. `.s` files belong to the base package vs xtest? I'm lacking Go knowledge here

Reply:
While working on cgo, I learned that cgo code is banned in test packages. `go list` has no way to analyze cgo in test packages since there are no fields to return it. Similar argument to be made for assembly since `go list` does not have `xtest_s_files` nor `test_s_files` keys.